### PR TITLE
[MSVC] work-around for compile time issue 102513

### PR DESCRIPTION
**Pull request by bd1976bris**
Disable optimizations when building clang/lib/AST/ByteCode/Interp.cpp with Microsoft's compiler as it has a bug that causes excessive build times. We do this only when NDEBUG is not defined on the assumption that building without asserts indicates that a user is strongly invested in runtime performance.

Partially addresses: https://github.com/llvm/llvm-project/issues/102513.

Once the bug is addressed in the Microsoft compiler this can be removed.

### CODE REVIEW

File: clang/lib/AST/ByteCode/Interp.cpp [hunk 1]
```diff
@@ -1430,6 +1434,10 @@ bool Interpret(InterpState &S, APValue &Result) {
     }
   }
 }
+// https://github.com/llvm/llvm-project/issues/102513
+#if defined(_MSC_VER)
```
Comment (rnk) [inline clang/lib/AST/ByteCode/Interp.cpp:1438]: Please make the ifdef conditions match, or define your own macro like DEOPTIMIZE_SWITCH. As written, I'd expect to see some clang-cl build warnings.

**Review by rnk — COMMENTED** [review]
Thanks!

**Comment by tbaederr** [comment]
Generally LGTM.

**Pull request merged**